src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1372,11 +1375,83 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     }
   }
 
-  function scheduleUpdate(fiber: Fiber, expirationTime: ExpirationTime) {
-    return scheduleUpdateImpl(fiber, expirationTime, false);
+  function scheduleUpdate(
+    fiber: Fiber,
+    partialState: mixed,
+    callback: (() => mixed) | null,
+    isReplace: boolean,
+    isForced: boolean,
+  ) {
```

Comment:
This is coupling too much into the scheduler. It seems nice now that we only have two types of updates but it'll come back to bite us later when we add more types. That's why as much as possible was carefully hoisted out into the class component.

In a WASM world this also mixes a bunch of JS-only objects that we couldn't pass into this module so we'd have to do it anyway.

Can we move this back into UpdateQueue and only keep the scheduleWork part here?